app/src/main/java/ru/spbhse/pocketmagic/NetworkController.java

```diff
@@ -0,0 +1,81 @@
+package ru.spbhse.pocketmagic;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+
+import java.nio.ByteBuffer;
+
+public class NetworkController {
+
+    private static AppCompatActivity currentActivity;
+    private static Controller currentController;
+    private static Network network;
+
+    public static Network createNetwork(GoogleSignInAccount account) {
```

Comment:
What if this is called twice? I think we would leak the first network. How about returning an existing network if it was already created, or throwing an exception if this is an invalid operation?

Reply:
Fixed.
Now if an Network object has already created, it returns the created one.